Makefile

```diff
@@ -0,0 +1,60 @@
+#
+# Usage
+#    make <script from package.json>
+#
+# Runs all tasks from package.json in correct order and deps, so this can be parallelized.
+# (Good results observed up to -j4).
+#
+# By default, all output of successfull command is skipped, to enable it set VERBOSE env variable
+# like this:
+#
+#    VERBOSE=1 make
+
+tasks = \
+	tslint prettier \
+	test \
+	build-examples \
+	build-bundle \
+	build-tests \
+	test-browser-chrome test-browser-firefox \
+	typedoc
+
+default: $(tasks)
+
+# custom targets and
+build: build-examples build-bundle
+test: test-browser
+test-browser: test-browser-chrome test-browser-firefox
+
+app_sources_ts=$(shell find @here/*/src -name "*.ts")
```

Comment:
Windows users start to hate you down from this line

Reply:
Yes, but windows users have MSYS2, Cygwin and WSL.
More, they have choice to not use this script and manually use `yarn whatever` in proper order.

`gradle`, `basel`, `cmake`, `ninja`, `Jake`, `grunt`, `gulp` based solutions are warmly welcomed but i seriously doubt that they can beat this one w.r.t robustness and simplicity.

Reply:
Yep, windows users have a lot of options not to use windows.

But seriously, make runs pretty good on plain windows, but there are some hard limitations:
 * no bash
 * no find

All other programs (like ```grep```, ```sed```, whatever) can be simply installed and used from default windows console.
But for ```bash``` and ```find``` you need to replace console: ```bash``` for obvious reasons, and ```find``` because windows has own builtin ```find``` command that can't be replaced.

So if we avoid this two conditions, we can run ```make``` on plain windows, if someone will need this feature of course.

Reply:
While it is fine to make it work in windows, if we can, our software does not work on plain windows already today. Some of our release scripts are already linux only. I would not take a too hard stand here if we have no alternative.

Reply:
If it doesn't work on Windows we should fix it. Introducing technology that we know doesn't work well on Windows is a no-no. I would like Windows developers to build our stack and also locally run the pre-submit checks so they can reproduce CI errors locally.

Reply:
> If it doesn't work on Windows

There are no major obstacles for it to run in proper environment on windows, just use MSYS2 - it should work. In GIT bash shell, it should work too.

@haraldF 
The point of this patch is: Improve developers experience by _enabling_ them to pre-check code in same way as in CI. CI works on Linux, more, it works only in Docker.

Counter question - can we run `.travis-yml` on Windows, Linux, OSX? No (at least not directly*), and yet it's here for it's purpose.

Let's not discard tools that help majority (majority work on Linux/Mac) given that minority has workarounds.

* [yes](https://andy-carter.com/blog/setting-up-travis-locally-with-docker-to-test-continuous-integration), we can but it's also, but we can run Makefile stuff in Docker too, it's just question of writing instruction

Reply:
@zbigg "no major obstacles" - I quite disagree :) It should work out of the box on a normal Windows terminal.

Plan B - what about having a simple task runner directly in node? So we can write `npm run ci:tests` and don't have any external tool dependencies (and no `sh` or `make` dependency)

Reply:
> Plan B - what about having a simple task runner directly in node? 

I tried hard to be pragmatic by not doing it by myself (DYI is my second name) so, namely using right tool for right job. As discussed internally, i still believe that `make` + `sh` - despite all of its limitations - has several qualities that qualify them. but ... ok.

Will rethink again proper task runner in bare TS (no gulp or other frameworks) for this, maybe it will "offspring" as proper "make" for JS projects.

Reply:
Maybe we can just write down some additional webpack rule (or yarn script) to make everything?